core/include/detray/core/detector_metadata.hpp

```diff
@@ -0,0 +1,250 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s)
+#include "detray/core/detail/multi_store.hpp"
+#include "detray/core/detail/single_store.hpp"
+#include "detray/definitions/containers.hpp"
+#include "detray/definitions/indexing.hpp"
+#include "detray/geometry/surface.hpp"
+#include "detray/intersection/cylinder_intersector.hpp"
+#include "detray/intersection/cylinder_portal_intersector.hpp"
+#include "detray/intersection/plane_intersector.hpp"
+#include "detray/masks/masks.hpp"
+#include "detray/masks/unbounded.hpp"
+#include "detray/masks/unmasked.hpp"
+#include "detray/materials/material_rod.hpp"
+#include "detray/materials/material_slab.hpp"
+#include "detray/surface_finders/accelerator_grid.hpp"
+#include "detray/surface_finders/brute_force_finder.hpp"
+
+// Covfie include(s)
+#include <covfie/core/backend/primitive/constant.hpp>
+#include <covfie/core/vector.hpp>
+
+namespace detray {
+
+/// mask-to-(next)-volume link (potentially switchable for SoA)
+using nav_link = std::uint_least16_t;
+
+/// mask types
+/// @TODO: Need to duplicate for pixel/strip measurement dimensions?
+using rectangle = mask<rectangle2D<>, nav_link>;
```

Comment:
Can we have those type definitions in metadata class or is it still better

Reply:
This is in the metadata file, or what to you mean exactly?

Reply:
I was meaning that we should avoid the global declarations

Reply:
Ah, Ok, got it. Yes, you are right